### PR TITLE
Tests, error

### PR DESCRIPTION
**Pull request by fyellin**
1.  Got rid of class in test_kernels
2.  Attempting to re-write test_errors forced me to re-examine error handling.
   a. Got rid of special handling of cspyce.sigerr().  It just fails, and the normal fail handling happens.
   b. cspice.sigerr() used to pop the chkin stack.  It no longer does this because doing that was a hack.  The functions in
        cspyce1.py are now responsible for dealing with checkin/checkout.
   c. A lot of the tests in test_error() presumed that sigerr popped the stack.  It no longer does that.
3. While I was in cspyce1.py, I got rid of all the .format statements.

### CODE REVIEW

**Comment by fyellin** [comment]
This is ready for review. . . .

File: cspyce/cspyce1.py [hunk 1]
```diff
@@ -199,13 +201,11 @@ def ckgp_error(inst, sclkdp, tol, ref):
...
 
-        chkin('ckgp_error')
-        setmsg('insufficient C kernel data to evaluate '
-               'instrument/spacecraft {}{} '
-               'at spacecraft clock time {} '
-               'with tolerance {}'.format(inst, namestr, sclkdp, tol))
-        sigerr('SPICE(CKINSUFFDATA)')
-        chkout('ckgp_error')
+        with chkin_and_chkout('ckgp_error'):
+            setmsg(f'insufficient C kernel data to evaluate instrument/spacecraft '
```
Comment (rfrenchseti) [inline cspyce/cspyce1.py:205]: Remove unneeded f

File: cspyce/cspyce1.py [hunk 2]
```diff
@@ -218,288 +218,248 @@ def ckgpav_error(inst, sclkdp, tol, ref):
...
 
-        chkin('ckgpav_error')
-        setmsg('insufficient C kernel data to evaluate '
-               'instrument/spacecraft {}{} '
-               'at spacecraft clock time {} '
-               'with tolerance {}'.format(inst, namestr, sclkdp, tol))
-        sigerr('SPICE(CKINSUFFDATA)')
-        chkout('ckgpav_error')
+        with chkin_and_chkout('ckgpav_error'):
+            setmsg(f'insufficient C kernel data to evaluate '
```
Comment (rfrenchseti) [inline cspyce/cspyce1.py:222]: Remove unneeded f

File: cspyce/cspyce1.py [hunk 3]
```diff
@@ -218,288 +218,248 @@ def ckgpav_error(inst, sclkdp, tol, ref):
...
-            chkin('gcpool_error')
-            setmsg('pool variable "{}" not found'.format(name))
-            sigerr('SPICE(VARIABLENOTFOUND)')
-            chkout('gcpool_error')
+    with chkin_and_chkout('gcpool_error'):
+        (cvals, found) = cspyce0.gcpool(name, start)
+        if not found:
+            ok, _count, _nctype = cspyce0.dtpool(name)
+            if not ok:
+                    setmsg(f'pool variable "{name}" not found')
```
Comment (rfrenchseti) [inline cspyce/cspyce1.py:271]: Bad indent

File: cspyce/cspyce1.py [hunk 4]
```diff
@@ -218,288 +218,248 @@ def ckgpav_error(inst, sclkdp, tol, ref):
...
-        ok, _count, _nctype = cspyce0.dtpool(name)
-        if not ok:
-            chkin('gcpool_error')
-            setmsg('pool variable "{}" not found'.format(name))
-            sigerr('SPICE(VARIABLENOTFOUND)')
-            chkout('gcpool_error')
+    with chkin_and_chkout('gcpool_error'):
+        (cvals, found) = cspyce0.gcpool(name, start)
+        if not found:
+            ok, _count, _nctype = cspyce0.dtpool(name)
```
Comment (rfrenchseti) [inline cspyce/cspyce1.py:269]: Why do some of these variables start with an underscore?

File: cspyce/cspyce1.py [hunk 5]
```diff
@@ -218,288 +218,248 @@ def ckgpav_error(inst, sclkdp, tol, ref):
...
+                return []
+
         (ok, count, nctype) = cspyce0.dtpool(name)
-        if not ok:
-            chkin('gdpool_error')
-            setmsg('pool variable "{}" not found'.format(name))
-            sigerr('SPICE(VARIABLENOTFOUND)')
-            chkout('gdpool_error')
+        if nctype != 'N':
+            setmsg(f'numeric values are not available; '
```
Comment (rfrenchseti) [inline cspyce/cspyce1.py:302]: Remove unneeded f

File: cspyce/cspyce1.py [hunk 6]
```diff
@@ -218,288 +218,248 @@ def ckgpav_error(inst, sclkdp, tol, ref):
...
+                return []
+
         (ok, count, nctype) = cspyce0.dtpool(name)
-        if not ok:
-            chkin('gipool_error')
-            setmsg('pool variable "{}" not found'.format(name))
-            sigerr('SPICE(VARIABLENOTFOUND)')
-            chkout('gipool_error')
+        if nctype != 'N':
+            setmsg(f'numeric values are not available; '
```
Comment (rfrenchseti) [inline cspyce/cspyce1.py:327]: Remove unneeded f

File: cspyce/cspyce1.py [hunk 7]
```diff
@@ -218,288 +218,248 @@ def ckgpav_error(inst, sclkdp, tol, ref):
...
-    if nctype != 'C':
-        setmsg('string values are not available; '
-               'kernel pool variable "{}" has numeric values'.format(item))
-        sigerr('SPICE(WRONGDATATYPE)')
-        chkout('stpool_error')
-        return ''
+            if nth != 0:
+                (_, ok) = cspyce0.stpool(item, 0, contin)
+                if ok:
+                    setmsg(f'index too large; '
```
Comment (rfrenchseti) [inline cspyce/cspyce1.py:432]: Remove unneeded f

File: cspyce/cspyce1.py [hunk 8]
```diff
@@ -734,13 +674,12 @@ def ckgp_vector_error(inst, sclkdp, tol, ref):
...
 
-        chkin('ckgp_vector_error')
-        setmsg('insufficient C kernel data to evaluate ' +
-               'instrument/spacecraft %s%s ' % (inst, namestr) +
-               'at spacecraft clock times %s ' % clockstr +
-               'with tolerance %s' % np.min(tol))
-        sigerr('SPICE(CKINSUFFDATA)')
-        chkout('ckgp_vector_error')
+        with chkin_and_chkout('ckgp_vector_error'):
+            setmsg(f'insufficient C kernel data to evaluate '
```
Comment (rfrenchseti) [inline cspyce/cspyce1.py:678]: Remove unneeded f

**Comment by fyellin** [comment]
Let me answer your questions.  If you decide you want me to fix, then I'll
fix. If you're okay with the code as it is (except for the bad line
indent), I'll leave the same.

There is some debate online as to how to express concatenated string
literals when some of them are f-strings and others aren't.  In PyCharm, If
you break up a long f-string into multiple literals, it automatically puts
an f at the front of every line, even if it's not necessary for that line.
I like this because it makes the strings line up nicely.  And internally,
it really is just one long f-string.

Regarding _obj.  Neither PyCharm nor PyLint complain about a local variable
not being used if it's name starts with _.  So rather than writing:
      name, _, state, country = address
I could write:
      name, _street, state, country = address
The advantages of the second form is that I'm giving the reader a hint at
what it is that's being skipped.  The code becomes simpler to modify in the
future.  If I need to look at the _address in a debugger, it's there.

I've seen this usage in other Python code.  Some commenters on Stack
Overflow love it, others hate it.  I assumed this is mentioned in some PEP,
but it really seems to be a thing added by PyLint and PyCharm.

I'll abide by whatever you decide.  You own the code.

**Comment by rfrenchseti** [comment]
For the leading underscore, I think I'm going to fall into the "hate it" camp. A leading underscore already has an accepted meaning in Python - it means "private". Like a private function or module-level variable that won't be imported. Giving it double-meaning of "don't complain if this variable isn't used" seems like a bad idea, and is lint-specific. And it's not mentioned in any PEP. I do understand the argument about making the code a little clearer and easier to modify in the future. But given the accepted meaning of a single leading underscore, I'm going to fall on the side of "just use _", which is universally accepted.

**Comment by rfrenchseti** [comment]
I will reverse my requests about the f strings with no substitute fields. It used to be whatever linter I was using would complain about those cases, but it seems that both pylint and flake8 no longer complain. So you can leave the f in to make it look pretty.

**Review by rfrenchseti — APPROVED** [review]

**Pull request merged**